### PR TITLE
[CDAP-2750] Ported JDBC driver cleanup fixes to DBSink

### PR DESCRIPTION
**Pull request by bdmogal**
- [x] Ported the JDBC driver fixes from #2966 to ``DBSink``. 
- [x] Fixed some exception handling.
- [x] Made the code less static so concurrent adapter runs function much better.

Build: http://builds.cask.co/browse/CDAP-RT39
Jira: [CDAP-2750](https://issues.cask.co/browse/CDAP-2750)

Currently running a test with SDK on Ubuntu.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java [hunk 1]
```diff
@@ -161,18 +168,16 @@ private void setResultSetMetadata(BatchSinkContext context) throws Exception {
...
-      // Load the plugin class to make sure it is available.
-      Class<?> driverClass = context.loadPluginClass(getJDBCPluginId());
-
       // Driver not found. We will try to register it with the DriverManager.
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
                 dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
                 JDBCDriverShim.class.getName());
-      DriverManager.registerDriver(new JDBCDriverShim((Driver) driverClass.newInstance()));
+      driverShim = new JDBCDriverShim(driverClass.newInstance());
+      DriverManager.registerDriver(driverShim);
```
Comment (poornachandra) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java]: We need to `DBUtils.deRegisterDriver(driverClass);` here too, right?

Comment (poornachandra) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java]: Also, maybe rename `deRegisterDriver` to `deRegisterAllDrivers`, or something better to reflect that all drivers with same classloader will get de-registered.?

File: cdap-common/src/main/java/co/cask/cdap/common/io/URLConnections.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Utility class for {@link URLConnection}.
+ */
+public class URLConnections {
+
+  /**
+   * Utility method to set {@link URLConnection#setDefaultUseCaches(boolean)}.
+   */
+  public static void setDefaultUseCaches(boolean useCaches) throws IOException {
+    final String currentDir = System.getProperty("user.dir");
+    new File(currentDir).toURI().toURL().openConnection().setDefaultUseCaches(useCaches);
```
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/common/io/URLConnections.java]: Need to close the connection with a finally.

Comment (poornachandra) [inline cdap-common/src/main/java/co/cask/cdap/common/io/URLConnections.java]: `URLConnection` does not have a `close` method.

**Comment by poornachandra** [comment]
LGTM

**Pull request merged**